src/appleseed/foundation/curve/binarycurvespecs.txt

```diff
@@ -0,0 +1,87 @@
+            Specifications of the BinaryCurve file format
+                    Revision 1 - May 28th, 2018
+
+
+
+INTRODUCTION
+
+  The purpose of the BinaryCurve file format is to store 3D curve geometry in a compact
+and efficient-to-read form.
+
+  The format is fully LITTLE-ENDIAN, regardless of the machine used to author
+files.
+
+
+
+GENERAL STRUCTURE
+
+  .----------------------------------.
+  |             Signature            |    11 bytes (string without 0 at the end)
+  +----------------------------------+
+  |              Version             |    2 bytes (16-bit unsigned integer)
+  +----------------------------------+
+  |               Data               |
+  `----------------------------------'
+
+  The signature field must contain the 11-character long string "BINARYCURVE".
+If it contains any other value, the file is not a valid BinaryCurve file.
+
+  The format of the Data block depends on the value of the Version field.
+
+
+
+DATA BLOCK FORMAT VERSION 1
+
+
+  .----------------------------------.
+  |             Basis                |    6 bytes (string without 0 at the end - linear/bezier/spline)
+  +----------------------------------+
+  |        Number of strands         |    2 bytes (16-bit unsigned integer)
```

Comment:
By `Number of strands`, you mean the total number of "curve" ?

And by `Number of vertices strand` you mean the number of vertices per curve ?

If `Number of strands` is 16 bits and `Number of vertices strand` is 32 bits, does it mean you can have more vertices per curve than the whole number of curve ? Looks weird to me.

Reply:
Yes. that is a mistake. I will correct it. Thanks.